### DISCUSSION TITLE
Why does analytics-browser clubs up events together in a single request?

### DISCUSSION THREAD

**Discussion by asharma-oanda** [amplitude/Amplitude-TypeScript#352] [Q&A]
I recently implemented @amplitude/analytics-browser in my react application. I have initialized the app and then I am using the `track` method to send the events to amplitude. However, I noticed there's a lot of duplication. When I load the app, it will show a request to `https://api2.amplitude.com/2/httpapi` with the request object firing the load event. When I click a button, it will again make a similar request but now it contains the load event as well as the click event. I was assuming that it will only send one event at a time but it looks like it keeps clubbing up the events and sends them to the server. Every time I perform an action, it clubs up with the previous events and sends it to the server. Is this something new in the SDK? How can I prevent it from sending all the events all over again and again?

Thanks.

**Comment by gnatyuk616** [comment]
@asharma-oanda hi!
I guess you could change flush & batch behavior: https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/#configure-batching-behavior

The described behavior looks like an implementation error. In Chrome's Networks and/or Amplitude Explorer plugin and/or Amplitude's User Look-up report you should expect 1 event per action.
